wolf/cmake/stream.cmake

```diff
@@ -125,25 +125,34 @@ if (WOLF_STREAM_RIST)
     
     set(FETCHCONTENT_QUIET OFF)
     FetchContent_MakeAvailable(${RIST_TARGET})
+
+    # this file should be available after building rist
+    set(RIST_LIBRARY_BINARY_FILE ${rist_BINARY_DIR}/librist.a)
```

Comment:
Do not specify the exact name of the expectation output file. This name may be changed in the original third party. Also, it is different in static or dynamic build types. Remove this line and work with the previous approach 'rist_command.out'.

Reply:
It's based on previous approach: https://github.com/WolfEngine/WolfEngine/pull/427/files/a80e34afe024365b3b75fcbb3072a66edd3cc471#diff-ce992633d6b2825f98b8f900dd57123bc6156c80a3d61c0f27ace68d2a002564L145

The change is mostly related to build dependency graph, so the specified file would be found, the rest is based on previous approach.

Reply:
Well, so leave your code unchanged, however, differentiate between static and dynamic build types. In other words, set `RIST_LIBRARY_BINARY_FILE ` to `librist.a` in static mode and `librist.so` in dynamic mode (Note please make sure that in dynamic building mode the generated library's name is `librist.so`).

Reply:
I'm not sure what you mean by "differentiate between static and dynamic build types" since we denote whether the library should be built as static or shared/dynamic.

I see that it was stated to be static both for windows (via meson `--default-library static` written in cmake) and android (via meson `--default-library static` in `librist-android.sh` file), which I did not modify and kept as is.

Reply:
As you mentioned, at the moment `RIST` library is only built in static mode, however, we may add the dynamic mode in the future. So, it's better to give the user the ability to specify the name of the output built file:

```
if(LIBRARY_TYPE MATCHES STATIC)
    set(RIST_LIBRARY_BINARY_FILE ${rist_BINARY_DIR}/librist.a)
else()
    set(RIST_LIBRARY_BINARY_FILE ${rist_BINARY_DIR}/librist.so)
endif()
```

Reply:
I'm confused, where does `LIBRARY_TYPE` come from?

and if it's a *might add*, I suppose this little change can be done at then?

Reply:
`LIBRARY_TYPE` is defined in the `wolf/CmakeLists.txt`. We can decide the building type of `RIST` library based on `wolf` building type. Anyway, leave this line unchanged as we can change and update it later.

Reply:
okay.